nrf_wifi/fw_if/umac_if/src/default/fmac_api.c

```diff
@@ -396,38 +394,23 @@ enum nrf_wifi_status nrf_wifi_fmac_dev_init(struct nrf_wifi_fmac_dev_ctx *fmac_d
 	}
 
 	nrf_wifi_osal_mem_set(fmac_dev_ctx->fpriv->opriv,
-			      rf_params,
-			      0xFF,
-			      sizeof(rf_params));
-
-	if (rf_params_usr) {
-		ret = nrf_wifi_utils_hex_str_to_val(fmac_dev_ctx->fpriv->opriv,
-						    rf_params,
-						    sizeof(rf_params),
-						    rf_params_usr);
+		             (unsigned char *)&phy_rf_params,
+		             0xFF,
+		             sizeof(phy_rf_params));
 
-		if (ret == -1) {
-			nrf_wifi_osal_log_err(fmac_dev_ctx->fpriv->opriv,
-					      "%s: hex_str_to_val failed",
-					      __func__);
-			status = NRF_WIFI_STATUS_FAIL;
-			goto out;
-		}
-	} else {
-		status = nrf_wifi_fmac_rf_params_get(fmac_dev_ctx,
-						     rf_params,
-						     tx_pwr_ceil_params);
+	status = nrf_wifi_fmac_rf_params_get(fmac_dev_ctx,
+                                             &phy_rf_params,
+					     tx_pwr_ceil_params);
 
-		if (status != NRF_WIFI_STATUS_SUCCESS) {
-			nrf_wifi_osal_log_err(fmac_dev_ctx->fpriv->opriv,
-					      "%s: RF parameters get failed",
-					      __func__);
-			goto out;
-		}
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
+		nrf_wifi_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+					"%s: RF parameters get failed",
+					__func__);
+		goto out;
 	}
 
 	status = nrf_wifi_fmac_fw_init(fmac_dev_ctx,
-				       rf_params,
+				       (unsigned char *)&phy_rf_params,
```

Comment:
Instead of casting here, the definition of `nrf_wifi_fmac_fw_init` can be changed to use `struct nrf_wifi_phy_rf_params`